### PR TITLE
Feat/alert capabilities

### PR DESCRIPTION
**Pull request by almogch**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
adding some new capabilities to the Intezer integration

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Intezer/Integrations/IntezerV2/IntezerV2.py [hunk 1]
```diff
@@ -486,6 +527,74 @@ def get_family_info_command(args: dict, intezer_api: IntezerApi) -> CommandResul
     )
 
 
+def submit_alert_command(args: dict, intezer_api: IntezerApi) -> CommandResults:
```
Comment (davidt99) [inline Packs/Intezer/Integrations/IntezerV2/IntezerV2.py:530]: Let's add optional source so we will be able to distinguish between xsoar and the original alert

File: Packs/Intezer/Integrations/IntezerV2/IntezerV2.py [hunk 2]
```diff
@@ -852,6 +962,166 @@ def enrich_dbot_and_display_endpoint_analysis_results(intezer_result, indicator_
...
+    )
+
+    indicator: Union[Common.Domain, Common.IP] | None = None
+
+    if artifact_analysis['artifact_type'] == 'domain':
+        indicator = Common.Domain(
+            domain=artifact_analysis['artifact_value'],
+            dbot_score=dbot_score
+        )
+    else:
```
Comment (davidt99) [inline Packs/Intezer/Integrations/IntezerV2/IntezerV2.py:1116]: be explict

File: Packs/Intezer/pack_metadata.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
     "name": "Intezer",
     "description": "Malware detection and analysis based on code reuse",
     "support": "partner",
-    "currentVersion": "1.5.11",
+    "currentVersion": "1.5.12",
```
Comment (davidt99) [inline Packs/Intezer/pack_metadata.json:5]: This probably needs to be 1.6

**Review by davidt99 — APPROVED** [review]

**Comment by maimorag** [comment]
Hi @almogch ,
Thank you for your contribution!
Is this pr is ready for review?

**Comment by almogch** [comment]
Hi @maimorag ,
Yes it is, 

thanks! :)

File: Packs/Intezer/ReleaseNotes/1_6_0.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#### Integrations
+##### Intezer v2
+- Updated the Docker image to: *demisto/py3-tools:1.0.0.67089*.
+- Added support for alert commands: 
+  - intezer-submit-alert
+  - intezer-submit-suspected-phishing-email
+  - intezer-get-alert-result
+- added 'related-alert-ids' param to 'intezer-analyze-by-file' command
```
Comment (maimorag) [inline Packs/Intezer/ReleaseNotes/1_6_0.md:8]:
```suggestion
- Added a 'related-alert-ids' parameter to 'intezer-analyze-by-file' command.
```

**Review by maimorag — COMMENTED** [review]
Good job @almogch!
Please add more unit tests so coverage would be at least 84.62% :)

File: Packs/Intezer/ReleaseNotes/1_6_0.md (hunk 4 above)
Comment (almogch) [reply Packs/Intezer/ReleaseNotes/1_6_0.md]: Done

**Comment by kgal-pan** [comment]
> Please add more unit tests so coverage would be at least 84.62% :)



@almogch You can see a report of the missed coverage in [CircleCI artifacts](https://output.circle-artifacts.com/output/job/31720e96-ae3f-4e1c-b880-7c58943cad1b/artifacts/0/artifacts/coverage_report/html/d_26e28e4beadd1dee_IntezerV2_py.html). 

One thing that we can see is that the `main` function is missed from coverage. We can add the `# pragma: no cover`:

```python
def main():  # pragma: no cover
```

**Comment by kgal-pan** [comment]
Seems that the UT and Lint jobs are failing because the Intezer SDK package cannot be found because the Intezer SDK dependency doesn't exist in the Docker images used for UT and lint. For example, in [this job](https://app.circleci.com/pipelines/github/demisto/content/288416/workflows/b6aaa4c2-9fd8-40c8-871d-cb1a50227f76/jobs/713949) we can see:

```
IntezerV2 - Get All Docker Images For Lint - Native image to run on is: demisto/py3-native:8.2.0.54663
```

We can see that https://github.com/demisto/dockerfiles/blob/master/docker/py3-native/pyproject.toml doesn't have the `intezer-sdk` dependency whereas the Docker image used in the integration, `py3-tools` does include it in https://github.com/demisto/dockerfiles/blob/master/docker/py3-tools/Pipfile so the UT and Lint [job using the `py3-tools` passes as expected](https://circleci.com/gh/demisto/content/713945).

@almogch - We're looking into this.

**Comment by kgal-pan** [comment]
> Seems that the UT and Lint jobs are failing because the Intezer SDK package cannot be found because the Intezer SDK dependency doesn't exist in the Docker images used for UT and lint. For example, in [this job](https://app.circleci.com/pipelines/github/demisto/content/288416/workflows/b6aaa4c2-9fd8-40c8-871d-cb1a50227f76/jobs/713949) we can see:
> 
> ```
> IntezerV2 - Get All Docker Images For Lint - Native image to run on is: demisto/py3-native:8.2.0.54663
> ```
> 
> We can see that https://github.com/demisto/dockerfiles/blob/master/docker/py3-native/pyproject.toml doesn't have the `intezer-sdk` dependency whereas the Docker image used in the integration, `py3-tools` does include it in https://github.com/demisto/dockerfiles/blob/master/docker/py3-tools/Pipfile so the UT and Lint [job using the `py3-tools` passes as expected](https://circleci.com/gh/demisto/content/713945).

https://jira-hq.paloaltonetworks.local/browse/CIAC-7655

**Comment by kgal-pan** [comment]
Upon further investigation, it seems like modules where the lint/UT are failing on:

```
/devwork/IntezerV2.py:562:16: E1101 submit_suspected_phishing_email_command: Class 'Alert' has no 'send_phishing_email' member
/devwork/IntezerV2.py:7:0: E0611 : No name 'alerts' in module 'intezer_sdk'
/devwork/IntezerV2.py:7:0: E0401 : Unable to import 'intezer_sdk.alerts'
/devwork/IntezerV2.py:12:0: E0611 : No name 'AlertInProgressError' in module 'intezer_sdk.errors'
/devwork/IntezerV2.py:13:0: E0611 : No name 'AlertNotFoundError' in module 'intezer_sdk.errors'
```

were only recently added in https://github.com/intezer/analyze-python-sdk/commit/d99e9f0294c9675dde21127008fa596ddf341ab5 as part of 1.18.4 release. The `py3-tools-ubi` (base image of py3-tools) image [uses Intezer SDK version 1.18.3](https://github.com/demisto/dockerfiles/blob/master/docker/py3-tools-ubi/poetry.lock#L448-L449) where the named modules don't exist yet. I'm working on fixing this.

File: Packs/Intezer/Integrations/IntezerV2/README.md [hunk 5]
```diff
@@ -6,12 +6,12 @@ Use the Intezer v2 integration to detect and analyze malware, based on code reus
 2. Search for Intezer v2.
 3. Click **Add instance** to create and configure a new integration instance.
 
-    | **Parameter** | **Description** | **Required** |
-    | --- | --- | --- |
-    | API Key |  | True |
-    | Intezer Analyze Base URL | The API address to intezer Analyze - i.e. https://analyze.intezer.com/api/ | False |
-    | Use system proxy settings |  | False |
-    | Trust any certificate (not secure) |  | False |
+   | **Parameter** | **Description** | **Required** |
```
Comment (guysl10) [inline Packs/Intezer/Integrations/IntezerV2/README.md:9]: Something looks weird here lets fix the table view

**Review by guysl10 — APPROVED** [review]

**Comment by almogch** [comment]
@kgal-pan thank you very much 🙏🏻

**Comment by maimorag** [comment]
Hi @almogch ,
Thank you for your contribution!
Good work :)

Few notes:
1. Update the docker image by running:
`demisto-sdk format -ud -i Packs/Intezer/Integrations/IntezerV2/IntezerV2.yml`
And the release note as well by running:
`demisto-sdk update-release-notes -i Packs/Intezer/Integrations/IntezerV2/IntezerV2.yml`

2. Please fix the unit tests' errors and pylint errors.

**Comment by almogch** [comment]
Hi @maimorag , I think I did all the things you suggested and I'm not sure what are the problems right now. 

From what I understand, the problem with the unit tests is still the issue with the docker image that isn't updated with the latest version of intezer-sdk version. 
Can you please have a look again and guide me on how to fix the remaining issues?

**Comment by kgal-pan** [comment]
@almogch - I believe you're correct, the UT/lint failures seen in https://circleci.com/gh/demisto/content/714643 are caused by an older version of the Intezer SDK. If I pull the image it's using in the `native:candidate` flow and look for the SDK packages, we can see that it has version 1.18.3 installed:

```bash
❯ docker run --rm -it  demisto/py3-native:8.3.0.67758 find /usr/local/lib/python3.10/site-packages -name "*intezer_sdk-*"

/usr/local/lib/python3.10/site-packages/intezer_sdk-1.18.3.dist-info
```

Looking into it.

**Comment by almogch** [comment]
@kgal-pan I just found out that on image `py3-native:8.3.0.68276` the intezer-sdk version is 1.18.6. which is perfect for me. I will try to upgrade the docker image on the PR and lets see if that fixes it

File: Tests/docker_native_image_config.json [hunk 6]
```diff
@@ -90,7 +90,7 @@
             "chromium",
             "tesseract"
          ],
-         "docker_ref": "demisto/py3-native:8.3.0.67758"
+         "docker_ref": "demisto/py3-native:8.3.0.68276"
```
Comment (maimorag) [inline Tests/docker_native_image_config.json:93]: Please revert the changes here.

**Review by maimorag — APPROVED** [review]

**Comment by almogch** [comment]
Hi @maimorag , no problem, I will fix it now.

The reason I changed it is that this is the last published image that contains the latest intezer-sdk package. 
After reverting it, the tests will probably fail again

**Comment by maimorag** [comment]
https://github.com/demisto/content/pull/28934 new pr was opened
